### PR TITLE
Expand native vim documentation

### PR DESCRIPTION
**Pull request by mandos**
Based on README.md

### CODE REVIEW

**Comment by dimbleby** [comment]
Thanks, it looks as though we have rather forgotten the docs!

I'm going to leave a couple of questions inline, which probably just reveal how little I know about writing docs in vim.

Although there's not very much, I'd also be happy to remove most of the duplication from the README and just have it point to the improved docs.  It seems as though we're not good at remembering to update two things...

File: doc/terraform.txt [hunk 1]
```diff
@@ -4,11 +4,44 @@ Author:  HashiVim <https://github.com/hashivim>
...
+
+                                      Type |gO| to see the table of contents.
+==============================================================================
+COMMANDS                                                 *terraform-commands*
 
 This command is only available if terraform is in your PATH.
 
-                                                *terraform-:terraform*
-:Terraform [args]         Invoke an arbitrary terraform command.
+                                                       *terraform-:terraform*
```
Comment (dimbleby) [inline doc/terraform.txt:14]: I realise this is how it was before, but shouldn't this be `*:Terraform*`?  `*terraform-:terraform*` looks like it's some sort of special syntax, what does it mean?

File: doc/terraform.txt [hunk 2]
```diff
@@ -4,11 +4,44 @@ Author:  HashiVim <https://github.com/hashivim>
...
+`:Terraform` [args]         Invoke an arbitrary terraform command.
+
+==============================================================================
+SETTINGS                                                 *terraform-settings*
+
+                                                        *'g:terraform_align'*
```
Comment (dimbleby) [inline doc/terraform.txt:20]: Why is this quoted?  Why not just `*g:terraform_align*`?  (obviously the same for the other settings)

I compared with a couple of random help files from other plugins and I don't see them quoting here.

File: doc/terraform.txt [hunk 3]
```diff
@@ -4,11 +4,44 @@ Author:  HashiVim <https://github.com/hashivim>
...
+`g:terraform_align`
```
Comment (dimbleby) [inline doc/terraform.txt:21]: Do we need this line?  Again I compared with a couple of other random help files and it looks as though they didn't.

File: doc/terraform.txt (hunk 1 above)
Comment (mandos) [reply doc/terraform.txt:14]: It's shows that :terraform belongs to plugin terraform, but I think `*:Terraform*` should be enough.

File: doc/terraform.txt (hunk 2 above)
Comment (mandos) [reply doc/terraform.txt:20]: This is convention to show that this is option of plugin, similar to `'filetype'`, `'ignorecase'` etc for vim, but I see that not many plugins use it, in my plugin's list only vim-go, I will remove quotes.

File: doc/terraform.txt (hunk 3 above)
Comment (mandos) [reply doc/terraform.txt:21]: I will leave only one.

**Comment by mandos** [comment]
Looks like documentation of Vim plugins is not very standardized. I will prepare another draft tomorrow.

**Comment by dimbleby** [comment]
This looks good to me, thank you very much!

Are you still making updates or is this ready to merge?

**Comment by mandos** [comment]
This part is ready for merge.

**Pull request merged**